sdk/keyvault/Azure.Security.KeyVault.Secrets.Extensions.Configuration/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -0,0 +1,170 @@
+            }
+
+            _pollingTask = null;
+            _cancellationToken = new CancellationTokenSource();
+            _reloadInterval = reloadInterval;
+        }
+
+        /// <summary>
+        /// Load secrets into this provider.
+        /// </summary>
+        public override void Load() => LoadAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task PollForSecretChangesAsync()
+        {
+            while (!_cancellationToken.IsCancellationRequested)
+            {
+                await WaitForReload().ConfigureAwait(false);
+                try
+                {
+                    await LoadAsync().ConfigureAwait(false);
+                }
+                catch (Exception)
+                {
+                    // Ignore
+                }
+            }
+        }
+
+        protected virtual Task WaitForReload()
+        {
+            // WaitForReload is only called when the _reloadInterval has a value.
+            return Task.Delay(_reloadInterval.Value, _cancellationToken.Token);
+        }
+
+        private async Task LoadAsync()
+        {
+            var secretPages = _client.GetPropertiesOfSecretsAsync();
+
+            var tasks = new List<Task<Response<KeyVaultSecret>>>();
+            var newLoadedSecrets = new Dictionary<string, LoadedSecret>();
+            var oldLoadedSecrets = Interlocked.Exchange(ref _loadedSecrets, null);
+
+            await foreach (var secret in secretPages.ConfigureAwait(false))
+            {
+                if (!_manager.Load(secret) || secret.Enabled != true)
+                {
+                    continue;
+                }
+
+                var secretId = secret.Name;
+                if (oldLoadedSecrets != null &&
+                    oldLoadedSecrets.TryGetValue(secretId, out var existingSecret) &&
+                    existingSecret.IsUpToDate(secret.UpdatedOn))
+                {
+                    oldLoadedSecrets.Remove(secretId);
+                    newLoadedSecrets.Add(secretId, existingSecret);
+                }
+                else
+                {
+                    tasks.Add(_client.GetSecretAsync(secret.Name));
+                }
+            }
+
+            await Task.WhenAll(tasks).ConfigureAwait(false);
+
+            foreach (var task in tasks)
+            {
+                var secretBundle = task.Result;
+                newLoadedSecrets.Add(secretBundle.Value.Name, new LoadedSecret(_manager.GetKey(secretBundle), secretBundle.Value.Value, secretBundle.Value.Properties.UpdatedOn));
+            }
+
+            _loadedSecrets = newLoadedSecrets;
+
+            // Reload is needed if we are loading secrets that were not loaded before or
+            // secret that was loaded previously is not available anymore
+            if (tasks.Any() || oldLoadedSecrets?.Any() == true)
```

Comment:
This boxes `true`. Check instead if `oldLoadedScrets` is null then check `Any()`. I also wrote an `IsNullOrEmpty()` extension method that may be in scope if you reference our shared project.

Reply:
true is not being boxed here, `oldLoadedSecrets?.Any() == true` is compiled to `list == null || !Enumerable.Any(list)`